### ISSUE TITLE
spotify.js

### ISSUE THREAD

**Issue by Axkaban** [hanord/ca-jammming#3]
Taking a look at Spotify.js makes me think that you have been working with JavaScript for a while, I saw my good friend `module.exports` on like 167. On your next react app I would recommend to lay back, relax and let babel hold your hand through all the modularity that comes with ES6+.

For example, something you can do is (and yes, it is the same as module.exports) create it as an object called Spotify and containing all the function methods like your `initAccessToken()` once you finished creating all the methods that you will use on app.js from spotify, you can do one line: `export default Spotify`.

Again, I want to say how much I love your thoroughness and your comments! (another think that made me think that you've been programming for a while). 

Here is something that I think you might find helpful, this is your work around initAccessToken:
[https://github.com/hanord/ca-jammming/blob/master/src/util/Spotify.js#L21-L102](https://github.com/hanord/ca-jammming/blob/master/src/util/Spotify.js#L21-L102)
Now some of those functions can be simplified for example timing, if token has expired it will certainly return empty (null) therfore in itself it can redirect to request access on the next call instead of timing its expiration, I do see the value of timing it though, it is very kind to a user. If you will like the solution version to get a token: 
`getAccessToken() {
		if (!accessToken) {
			//if access token doesnt' exist, check the URL to see if there is a token in it
			const url_string = window.location.href;

			const a = url_string.match(/access_token=([^&;]*)/);
			const e = url_string.match(/expires_in=([^&]*)/);

			//if there is a token in the url, then use that value
			if (a && e) {
				accessToken = a[1];
				expiry = e[1];		

				window.setTimeout(() => accessToken = '', expiry * 1000);
				window.history.pushState('Access Token', null, '/')		

			}
			//if not, redirect to spotify to request access
			else {
				window.location.href = `${authURL}?client_id=${clientId}&response_type=token&redirect_uri=${redirURL}&scope=${scopes}`;

			}
		}
		
		return accessToken;
	},

	search(term) {

		const accessToken = Spotify.getAccessToken();
		const url = `${corsURL}/${apiURL}/search?type=track&q=${term}`;

		return fetch(url, {
			headers: { Authorization: `Bearer ${accessToken}` },
		})
		.then(response => response.json())
		.then(jsonResponse => {
			if (jsonResponse.tracks) {
				return jsonResponse.tracks.items.map(track => {
					return {
						id: track.id,
						title: track.name,
						author: track.artists[0].name,
						album: track.album.name,
						uri: track.uri
					}
				})
			}
			else {
				return [];
			}
		})
	}`

**Comment by hanord** [comment]
Great stuff. I'll look at this and try out your suggestions and get back in a couple of days. And yes, I used to write software many years ago. Back in the 90s I invented a C++ software framework called Qt (still relevant today; qt.io) but I've been focusing on running companies the last 20 years. Now I'm building two startup companies and I'd like to work closely with our tech teams and maybe even help out with some coding. So I'm trying a teach myself to become a fullstack developer with React, Go and microservices. There's a lot to learn but I enjoy spending late hours coding again.
